framework/encode/vulkan_capture_common.cpp

```diff
@@ -0,0 +1,263 @@
+{
+#if defined(VK_USE_PLATFORM_ANDROID_KHR)
+    if (vulkan_capture_manager && !vulkan_capture_manager->IsCaptureModeWrite())
+    {
+        return;
+    }
+    assert(hardware_buffer != nullptr);
+
+    format::CreateHardwareBufferCommandHeader create_buffer_cmd;
+
+    create_buffer_cmd.meta_header.block_header.type = format::BlockType::kMetaDataBlock;
+    create_buffer_cmd.meta_header.block_header.size = format::GetMetaDataBlockBaseSize(create_buffer_cmd);
+    create_buffer_cmd.meta_header.meta_data_id      = format::MakeMetaDataId(
+        format::ApiFamilyId::ApiFamily_Vulkan, format::MetaDataType::kCreateHardwareBufferCommand);
+    create_buffer_cmd.thread_id = thread_id;
+    create_buffer_cmd.memory_id = memory_id;
+    create_buffer_cmd.buffer_id = reinterpret_cast<uint64_t>(hardware_buffer);
+
+    // Get AHB description data.
+    AHardwareBuffer_Desc ahb_desc = {};
+    AHardwareBuffer_describe(hardware_buffer, &ahb_desc);
+
+    create_buffer_cmd.format = ahb_desc.format;
+    create_buffer_cmd.width  = ahb_desc.width;
+    create_buffer_cmd.height = ahb_desc.height;
+    create_buffer_cmd.stride = ahb_desc.stride;
+    create_buffer_cmd.usage  = ahb_desc.usage;
+    create_buffer_cmd.layers = ahb_desc.layers;
+
+    size_t planes_size = 0;
+
+    if (plane_info.empty())
+    {
+        create_buffer_cmd.planes = 0;
+    }
+    else
+    {
+        create_buffer_cmd.planes = static_cast<uint32_t>(plane_info.size());
+        // Update size of packet with compressed or uncompressed data size.
+        planes_size = sizeof(plane_info[0]) * plane_info.size();
+        create_buffer_cmd.meta_header.block_header.size += planes_size;
+    }
+
+    if (vulkan_capture_manager)
+    {
+        if (planes_size > 0)
+        {
+            vulkan_capture_manager->CombineAndWriteToFile(
+                { { &create_buffer_cmd, sizeof(create_buffer_cmd) }, { plane_info.data(), planes_size } });
+        }
+        else
+        {
+            vulkan_capture_manager->WriteToFile(&create_buffer_cmd, sizeof(create_buffer_cmd));
+        }
+    }
+    else if (vulkan_state_writer)
+    {
+        vulkan_state_writer->OutputStreamWrite(&create_buffer_cmd, sizeof(create_buffer_cmd));
+
+        if (planes_size > 0)
+        {
+            vulkan_state_writer->OutputStreamWrite(plane_info.data(), planes_size);
+        }
+
+        vulkan_state_writer->IncrementBlocksWritten();
+    }
+#else
+    GFXRECON_UNREFERENCED_PARAMETER(thread_id);
+    GFXRECON_UNREFERENCED_PARAMETER(memory_id);
+    GFXRECON_UNREFERENCED_PARAMETER(hardware_buffer);
+    GFXRECON_UNREFERENCED_PARAMETER(plane_info);
+    GFXRECON_UNREFERENCED_PARAMETER(vulkan_capture_manager);
+    GFXRECON_UNREFERENCED_PARAMETER(vulkan_state_writer);
+
+    GFXRECON_LOG_ERROR("Skipping create AHardwareBuffer command write for unsupported platform");
+#endif
+}
+
+static void CommonWriteFillMemoryCmd(format::HandleId      memory_id,
```

Comment:
I think it would be better if instead of defining this function to pass the pointer to Writer's/Manager's `WriteFillMemoryCmd` as an additional argument to `CommonProcessHardwareBuffer`. But this would require writing a declaration for that function pointer somewhere that is common to all managers and is api agnostic since this function is referred to from a lot of places (dx12 capture manager,  vulkan capture manager and state writer, and the page guard manager).

I think this gets outside of the scope of this PR so I'll just leave this here as a suggestion for a different PR.